storage-proofs/src/election_post.rs

```diff
@@ -481,18 +486,40 @@ mod tests {
 
             let graph = BucketGraph::<H>::new(leaves, BASE_DEGREE, 0, new_seed()).unwrap();
 
+            let replica_path = temp_path.join(format!("replica-path-{}", i));
+            let mut f = File::create(&replica_path).unwrap();
+            f.write_all(&data).unwrap();
+
             let cur_config =
                 StoreConfig::from_config(&config, format!("test-lc-tree-v1-{}", i), None);
             let mut tree: QuadMerkleTree<_, _> = graph
                 .merkle_tree(Some(cur_config.clone()), data.as_slice())
                 .unwrap();
             let c = tree
-                .compact(cur_config.clone(), StoreConfigDataVersion::One as u32)
+                .compact(cur_config.clone(), StoreConfigDataVersion::Two as u32)
                 .unwrap();
             assert_eq!(c, true);
 
+            // Open the replica for base layer data reads.
+            let reader = OpenOptions::new()
+                .read(true)
+                .open(&replica_path)
+                .expect("Failed to open replica (base layer data)");
+
+            // Configure an external reader for access to the base layer data.
+            let external_reader = ExternalReader {
```

Comment:
this `ExternalReader` is always the same, I would suggest moving this impl into `.lcmerkle_tree` and just give it the path.